docs/config/dns.md

```diff
@@ -1,13 +1,57 @@
 # DNS 域名解析
 
-V2Ray 内建了一个 DNS 模块，其主要用途为：对目标地址（域名）进行 DNS 解析，同时为 IP 路由规则匹配提供判断依据。
+V2Ray 内建了一个 DNS 组件，其主要用途为：对目标地址（域名）进行 DNS 解析，同时为 IP 路由规则匹配提供判断依据。
 
 :::tip
 由于 DNS 协议的复杂性，V2Ray 只支持最基本的 IP 查询（A 和 AAAA 记录）。如需完整的 DNS 功能，推荐使用 [CoreDNS](https://coredns.io)。
 :::
 
 :::warning
-在 `freedom` 协议的 `outbound` 中，`domainStrategy` 默认值为 `AsIs`，不会使用本 DNS 模块进行目标地址解析。如需使用，应配置为 `UseIP`、`UseIPv4` 或 `UseIPv6`。
+在 `freedom` 协议的 `outbound` 中，`domainStrategy` 默认值为 `AsIs`，即 `freedom` outbound 默认不会使用本 DNS 组件进行目标地址解析。如需使用，应修改 `freedom` outbound 中的 `domainStrategy` 配置为 `UseIP`、`UseIPv4` 或 `UseIPv6`。
+:::
+
+## 支持的 DNS 协议及其路由策略
+
+- DNS over **UDP**：查询请求经过路由组件，可从用户指定 outbound 发出
+  - 使用 IP 地址的形式，如 `8.8.8.8`
+  - 默认使用 `53` 端口进行查询，支持使用非标准端口
+- DNS over **TCP**：查询请求经过路由组件，可从用户指定 outbound 发出
+  - 使用 `tcp://host:port` 的形式，如 `tcp://8.8.8.8:53`
+  - 默认使用 `53` 端口进行查询，支持使用非标准端口
+  - v4.40.0 及以上版本可使用
+- DNS over **TCP local mode**：查询请求不经过路由组件，直接从 freedom outbound 发出
+  - 使用 `tcp+local://host:port` 的形式，如 `tcp+local://8.8.8.8:53`
+  - 默认使用 `53` 端口进行查询，支持使用非标准端口
+  - v4.40.0 及以上版本可使用
+- DNS over **HTTPS**：查询请求不经过路由组件，直接从配置文件中的第一个 outbound 发出
```

Comment:
Typo here.

Reply:
Where is it? I can't find it.

Reply:
Sorry, I added the comment on a single line, but somehow it became 4 lines. DoH traffic actually goes through the router.

Reply:
Confirmed that non-local DoH traffic goes through the router.

Reply:
Does it go through the router, and choose the first outbound? 😂

Reply:
@Loyalsoldier No, the previous documentation was incorrect. The tests I conducted showed that non-local DoH traffic respects routing rules and is not locked to the first outbound.

```
2021/08/10 01:08:37 [Info] app/dns: DOH//1.1.1.1 querying: api6.ipify.org.
2021/08/10 01:08:37 [Info] app/dispatcher: taking detour [not-the-first-outbound] for [tcp:1.1.1.1:443]
2021/08/10 01:08:37 [Info] app/dispatcher: taking detour [not-the-first-outbound] for [tcp:1.1.1.1:443]
2021/08/10 01:08:37 [Info] transport/internet/tcp: dialing TCP to tcp:192.168.50.18:1085
2021/08/10 01:08:37 [Info] transport/internet/tcp: dialing TCP to tcp:192.168.50.18:1085
```

The TCP target it dialed matches the selected outbound, which is not the first outbound and is selected by the routing rules.

Reply:
What about using `https://dns.google` and do NOT set routing rule or host for it?

Reply:
With `https://cloudflare-dns.com/dns-query` as the only DNS server:

```
2021/08/10 11:07:32 [Info] [2870114501] proxy/socks: TCP Connect request to tcp:api6.ipify.org:443
2021/08/10 11:07:32 [Info] app/dns: DOH//cloudflare-dns.com querying: api6.ipify.org.
2021/08/10 11:07:32 [Info] app/dispatcher: taking detour [last-but-not-least-outbound] for [tcp:cloudflare-dns.com:443]
2021/08/10 11:07:32 [Info] transport/internet/tcp: dialing TCP to tcp:192.168.50.18:1085
2021/08/10 11:07:32 [Info] app/dispatcher: taking detour [last-but-not-least-outbound] for [tcp:cloudflare-dns.com:443]
2021/08/10 11:07:32 [Info] transport/internet/tcp: dialing TCP to tcp:192.168.50.18:1085
```

The detour is caused by the last rule in my `IPOnDemand` routing config:

```json
            {
                "type": "field",
                "network": "tcp,udp",
                "outboundTag": "last-but-not-least-outbound"
            }
```

It should be noted in the documentation that IP rules do not affect DoH traffic from domain-based DoH servers. They do apply to DoH traffic from IP-based DoH servers though (e.g. https://1.1.1.1/dns-query). Domain rules apply to DoH traffic.

Reply:
Fixed by https://github.com/v2fly/v2fly-github-io/pull/221